### PR TITLE
[SYM-4800] Update Tailscale Strategy example to use runtime-connector module 2.x

### PR DESCRIPTION
**Pull request by Arianna2028**
## Description

- Updates the `basic/tailscale_ssh_access` example to use version 2.x of the [runtime-connector](https://registry.terraform.io/modules/symopsio/runtime-connector/aws/latest) module.
- Updates the usages of `main` to reference the `local.environment_name` variable instead, for consistency.
- Updates the Sym Terraform Provider to v3 and the `sym_flow.implementation` to be file contents, which is required in v3.
- Removes `runtime_id` from the `sym_environment`, since it is not needed.

## Testing

Tested using the alpha version of the runtime-connector module and the following `override.tf`:
```hcl
provider "sym" {
  org = "sym"
}

module "runtime_connector" {
  source  = "symopsio/runtime-connector/aws"
  version = "2.0.0-alpha"

  environment_name = local.environment_name
  sym_account_id = "906191524257"
}

resource "sym_integration" "tailscale" {
  type        = "tailscale"
  name        = "${local.environment_name}-tailscale-integration"
  external_id = "symops.io"

  settings = {
    api_token_secret = sym_secret.tailscale_api_key.id
  }
}
```

### CODE REVIEW

**Comment by Arianna2028** [comment]
Current dependencies on/for this PR:
* main
  * **PR #69** <a href="https://app.graphite.dev/github/pr/symopsio/examples/69" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/symopsio/examples/69?utm_source=stack-comment).

File: basic/tailscale_ssh_access/secrets.tf [hunk 1]
```diff
@@ -23,7 +23,7 @@ EOT
 
 resource "aws_iam_role_policy_attachment" "attach_secrets_manager_access" {
```
Comment (llam15) [inline basic/tailscale_ssh_access/secrets.tf:24]: Let's use the secrets manager module here

File: basic/tailscale_ssh_access/tailscale_ssh_access.tf [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # An AWS Secrets Manager Secret to hold your Tailscale API Key. Set the value with:
-# aws secretsmanager put-secret-value --secret-id "main/tailscale-api-key" --secret-string "YOUR-TAILSCALE-API-KEY"
+# aws secretsmanager put-secret-value --secret-id "${local.environment_name}/tailscale-api-key" --secret-string "YOUR-TAILSCALE-API-KEY"
 resource "aws_secretsmanager_secret" "tailscale_api_key" {
-  name        = "main/tailscale-api-key"
+  name        = "${local.environment_name}/tailscale-api-key"
```
Comment (llam15) [inline basic/tailscale_ssh_access/tailscale_ssh_access.tf:4]:
```suggestion
  name        = "sym/${local.environment_name}/tailscale-api-key"
```

Let's prefix the secret here too

File: basic/tailscale_ssh_access/tailscale_ssh_access.tf [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
   description = "API Key for Sym to call Tailscale APIs"
 
-  # This SymEnv tag is required and MUST match the SymEnv tag in the 
+  # This SymEnv tag is required and MUST match the SymEnv tag in the
   # aws_iam_policy.secrets_manager_access in your `secrets.tf` file
```
Comment (llam15) [inline basic/tailscale_ssh_access/tailscale_ssh_access.tf:8]: After updating `secrets.tf` to use the module, let's update this comment

**Review by llam15 — APPROVED** [review]

**Pull request merged**